sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -385,6 +400,9 @@ def test_check_estimator():
     check_estimator(MultiTaskElasticNet)
     check_estimator(MultiTaskElasticNet())
 
+    # doesn't error on binary_only tagged estimator
+    check_estimator(BinaryDecisionTreeClassifier)
```

Comment:
Can you assert that it raises an error if the tag is not present? Otherwise looks good.
Might be worth considering a method for creating the data like for multi-output and pairwise estimators. But maybe we can leave it as-is for now and later consolidate once we figured out how to actually deal with different data types.

Reply:
Sure, done @amueller 

Looks like we're green on tests now